browser/extensions/BUILD.gn

```diff
@@ -8,6 +8,10 @@ source_set("extensions") {
     "api/brave_rewards_api.h",
     "api/brave_shields_api.cc",
     "api/brave_shields_api.h",
+    "api/brave_sync_api.cc",
+    "api/brave_sync_api.h",
+    "api/brave_sync_event_router.cc",
```

Comment:
I think we need to find more better place for router.cc in the future because it's not extension api.